src/config/context.rs

```diff
@@ -85,6 +84,8 @@ pub fn build_object_store(
             bucket,
             cache_properties,
         }) => {
+            // Bucket from the config may contain some path so extract the proper bucket
+            let bucket = bucket.split('/').next().unwrap_or(bucket);
```

Comment:
I went with overloading the bucket config param so that it can also hold a path to stay backward-compatible.

But I think the preferable way to do this is to ditch the `type` and `bucket` fields and replace them with a `location` field which can store all the relevant info (i.e. type, bucket and optionally path) at the expense of breaking backward-compatibility.

Reply:
Could keep the `bucket` and have a separate `prefix` field that is optional and defaults to no prefix where it doesn't exist?

Reply:
I thought that a separate `prefix` field wasn't worth the increased config complexity, whereas the `location` approach felt more consolidated than either alternative.

To me it seems like it's natural to think about the full url `location` as one thing from the user perspective, and leave the parsing of the individual details (type, bucket, path) to Seafowl. This would also neatly get rid of the `endpoint` field on S3, as this is covered by `location` too. Moreover, currently we don't support passing the bucket/path in the `endpoint`, though in principle there's no reason not to. [GlareDB uses](https://github.com/GlareDB/glaredb/blob/b9cea63d1326e4528be73bd154cd13f16637302d/.github/workflows/ci.yaml#L315-L319) the `location` + storage options map approach and does away with config file completely due to the conciseness of this approach, though this wouldn't work for Seafowl as we have a bunch of other configurations as well.

To clarify, I definitely do prefer a separate `path` (or `prefix`) field over overloading the `bucket` field, I just think `location` would be even better from a usability standpoint. Let me know what are your preferences.

Reply:
Alright, I made the change to use the `prefix` field for now as that's good enough for a merge I believe, but we can revisit whether it makes sense to go with the `location` approach later on.